domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/CpuPerformanceSnapshotter.kt

```diff
@@ -88,7 +96,10 @@ class CpuPerformanceSnapshotter(
       }
     }.also {
       CoroutineScope(backgroundCoroutineDispatcher).launch {
-        sendTakeSnapshotCommand(switchIconificationCount)
```

Comment:
How does this behave if we don't create an initial snapshot at the outset? I wonder if the UNINITIALIZED branch should actually be part of the public snapshot function since otherwise the snapshotter might behave a little weird if this is started without a snapshot. Or, does it already work that way (if the queue is lazy)?

These are good initialization tests to add, if they don't already exist, e.g.:
- Snapshotter does nothing if no snapshot is sent, both immediately, after the iconification delay, after fg delay, and after bg delay
- takeSnapshot is required to initiate the iconification initialization timer

Reply:
I've modified the snapshotter a little bit to accommodate the `logging-once-initialised` case. Essentially, a new boolean variable was needed to determine if the snapshotter was initialised or not -- this was done since an update iconification call with a fg/bg switch would've initiated logging and hence a second variable was required to keep a check on the initialisation status that determines if a previous snapshot exists or not. 

I've also written test cases covering the above examples that you mentioned along with a few more that I thought of on the basis of the above change. To clarify, I've not added a specific test for the second example, since it gets tested with `testSnapshotter_onCreate_setsIconificationToBgAfterCutOff_logsCpuInBgAfterCorrectDelay`.

Reply:
I thought of removing the lazy implementation completely and introduce an initialisation method. It was based on the assumption that we can rule out `commandQueue` and use actor channels instead. But it wasn't entirely possible since the `commandQueue` was getting used in methods that are not present in the actor and hence don't have access to the channel instance -- for example: `sendSwitchIconificationCommand` in `updateAppIconification` doesn't have access to the channel and hence can't use the same actor instance to perform the update.

Reply:
Thanks--the tests look really good.